### PR TITLE
Add a re-indexer service for resources indexed into Orangelight

### PR DESCRIPTION
**Pull request by eliotjordan**

### CODE REVIEW

File: app/services/orangelight_reindexer.rb [hunk 1]
```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+class OrangelightReindexer
+  # Reindexes resources indexed into Orangelight by sending record updated messages
+  # for all complete resources.
+  # @param optional logger [Logger] instance of logger class
+  def self.reindex_orangelight(logger: Logger.new(STDOUT))
+    new(logger: logger).reindex_orangelight
+  end
+
+  attr_reader :logger, :ogm_repo_path
```
Comment (tpendragon) [inline app/services/orangelight_reindexer.rb]: `ogm_repo_path` left over? Can you remove it?

File: app/services/orangelight_reindexer.rb [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  private
+
+    def all_orangelight_resources
+      [
+        complete_resources(model: Numismatics::Coin)
+      ]
+    end
+
+    def complete_resources(model:)
+      resources = query_service.find_all_of_model(model: model)
```
Comment (tpendragon) [inline app/services/orangelight_reindexer.rb]: This is going to be really slow, right? If I spun up a PR real quick to add model filtering and lazy-loaded cursors to `find_by_property` would you use that instead?

File: app/services/orangelight_reindexer.rb [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+  end
+
+  attr_reader :logger, :ogm_repo_path
+  def initialize(logger:)
+    @logger = logger
+  end
+
+  def reindex_orangelight
+    all_orangelight_resources.each do |resources|
+      resources.each do |resource|
```
Comment (tpendragon) [inline app/services/orangelight_reindexer.rb]: Why are there two arrays?

File: lib/tasks/reindex.rake [hunk 4]
```diff
@@ -28,3 +28,10 @@ namespace :geoblacklight do
     GeoResourceReindexer.reindex_geoblacklight
   end
 end
+
+namespace :orangelight do
+  desc "Issue rabbitmq messages that each complete orangelight resource is updated"
+  task reindex: :environment do
+    OrangelightReindexer.reindex_orangelight
```
Comment (tpendragon) [inline lib/tasks/reindex.rake]: Maybe you should force `ENV["BULK"] = true` here if it's important, and test for it in the message.

Comment (hackmastera) [reply lib/tasks/reindex.rake]: I like this idea

File: app/services/orangelight_reindexer.rb (hunk 2 above)
Comment (tpendragon) [reply app/services/orangelight_reindexer.rb]: https://github.com/pulibrary/figgy/pull/3756 has that feature.

Comment (hackmastera) [reply app/services/orangelight_reindexer.rb]: incorporated

File: app/services/orangelight_reindexer.rb (hunk 3 above)
Comment (hackmastera) [reply app/services/orangelight_reindexer.rb]: it was copied from the geo service. I un-nested them.

File: lib/tasks/reindex.rake (hunk 4 above)
Comment (hackmastera) [reply lib/tasks/reindex.rake]: @tpendragon I forced it in the service. Is 1b10d38 what you were thinking?

Comment (tpendragon) [reply lib/tasks/reindex.rake]: Looks good to me.

**Pull request merged**